src/org/labkey/test/pages/core/admin/AllowedFileExtensionAdminPage.java

```diff
@@ -0,0 +1,189 @@
+package org.labkey.test.pages.core.admin;
+
+import org.labkey.test.BootstrapLocators;
+import org.labkey.test.Locator;
+import org.labkey.test.Locators;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.WebTestHelper;
+import org.labkey.test.components.html.Input;
+import org.labkey.test.pages.LabKeyPage;
+import org.labkey.test.util.PortalHelper;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AllowedFileExtensionAdminPage extends LabKeyPage<AllowedFileExtensionAdminPage.ElementCache>
+{
+    public AllowedFileExtensionAdminPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    public static AllowedFileExtensionAdminPage beginAt(WebDriverWrapper webDriverWrapper)
+    {
+        webDriverWrapper.beginAt(WebTestHelper.buildURL("admin", "allowList"));
+        return new AllowedFileExtensionAdminPage(webDriverWrapper.getDriver());
+    }
+
+    @Override
+    protected void waitForPage()
+    {
+        waitFor(()-> {
+            try
+            {
+                return !BootstrapLocators.loadingSpinner.areAnyVisible(getDriver()) &&
+                        elementCache().extension.getComponentElement().isDisplayed();
+            }
+            catch (NoSuchElementException | StaleElementReferenceException | TimeoutException retry)
+            {
+                return false;
+            }
+        }, "Allowed File Extensions page did not load in time.", 3_000);
+
+    }
+
+    public AllowedFileExtensionAdminPage setExtension(String extension)
+    {
+        elementCache().extension.set(extension);
+        return this;
+    }
+
+    public String clickSaveExpectingError()
+    {
+        String errorText;
+        elementCache().saveExtension.click();
+        errorText = waitForElement(Locators.labkeyError).getText();
+        clearCache();
+        return errorText;
+    }
+
+    public AllowedFileExtensionAdminPage clickSaveExtension()
+    {
+        elementCache().saveExtension.click();
+        clearCache();
+        return this;
+    }
```

Comment:
All of the buttons (delete, save, and update) trigger a page load, regardless of whether there is an error. These should all wait for a page load and error-free versions should verify that there is no error.
```suggestion
    private AllowedFileExtensionAdminPage clickButtonNoError(WebElement button)
    {
        clickAndWait(button);
        clearCache();
        assertNoLabKeyErrors();
        return this;
    }

    private String clickButtonExpectingError(WebElement button)
    {
        clickAndWait(button);
        clearCache();
        return waitForElement(Locators.labkeyError).getText();
    }

    public String clickSaveExpectingError()
    {
        return clickButtonExpectingError(elementCache().saveExtension);
    }

    public AllowedFileExtensionAdminPage clickSaveExtension()
    {
        return clickButtonNoError(elementCache().saveExtension);
    }
```

And so on for update and delete.

Reply:
Updated.